From 8bd34d5e1a44a179024fcda6d48ba7c30f753f08 Mon Sep 17 00:00:00 2001
From: Jianing Ren <jianing.ren@rock-chips.com>
Date: Fri, 1 Nov 2019 17:54:02 +0800
Subject: [PATCH] arm64: dts: rk3399: fix the extcon property of usb

In kernel 4.4, we use the extcon notification function in dwc3-rockchip.c
to perform the drd mode switch. Kernel 4.19 implements the same
functionality in drd.c and writes the extcon notification function in
core.c. So we need to move extcon to the subnode of dwc3.

Besides, I deleted some useless code which is wrongly copied.

Change-Id: I1096e64d5a29f01c94a10027b846676033c7985d
Signed-off-by: Jianing Ren <jianing.ren@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-box.dtsi           |  2 +-
 arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi       |  2 +-
 arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi           |  4 ++--
 .../arm64/boot/dts/rockchip/rk3399-firefly-android.dts |  2 +-
 arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts  |  2 +-
 arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi           |  2 +-
 .../arm64/boot/dts/rockchip/rk3399-mid-818-android.dts |  2 +-
 arch/arm64/boot/dts/rockchip/rk3399-rock960-ab.dts     |  2 +-
 arch/arm64/boot/dts/rockchip/rk3399-rockpro64.dts      |  2 +-
 arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi      | 10 +---------
 arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi      |  2 +-
 arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts       |  4 ++--
 .../boot/dts/rockchip/rk3399pro-evb-v10-linux.dts      |  2 +-
 arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10.dts     |  2 +-
 .../boot/dts/rockchip/rk3399pro-evb-v11-linux.dts      |  2 +-
 arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11.dts     |  2 +-
 16 files changed, 18 insertions(+), 26 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
index 3ff1c59a0c35..d264ece43f93 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
@@ -626,13 +626,13 @@
 };
 
 &usbdrd3_0 {
-	extcon = <&fusb0>;
 	status = "okay";
 };
 
 &usbdrd_dwc3_0 {
 	dr_mode = "otg";
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usbdrd3_1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi
index 7eb70d7985c0..07da6487b545 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi
@@ -1184,7 +1184,6 @@
 
 &usbdrd3_0 {
 	status = "okay";
-	extcon = <&fusb0>;
 };
 
 &usbdrd3_1 {
@@ -1193,6 +1192,7 @@
 
 &usbdrd_dwc3_0 {
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usbdrd_dwc3_1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
index 3405d392e237..b321d1e76e97 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb.dtsi
@@ -490,21 +490,21 @@
 };
 
 &usbdrd3_0 {
-	extcon = <&fusb0>;
 	status = "okay";
 };
 
 &usbdrd_dwc3_0 {
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usbdrd3_1 {
-	extcon = <&fusb1>;
 	status = "okay";
 };
 
 &usbdrd_dwc3_1 {
 	status = "okay";
+	extcon = <&fusb1>;
 };
 
 &pwm0 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
index 3d96d655d720..9a992991a1a9 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
@@ -1054,7 +1054,6 @@
 
 &usbdrd3_0 {
 	status = "okay";
-	extcon = <&fusb0>;
 };
 
 &usbdrd3_1 {
@@ -1063,6 +1062,7 @@
 
 &usbdrd_dwc3_0 {
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usbdrd_dwc3_1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
index 82bb2a63fe73..53725c338fa7 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
@@ -1028,7 +1028,6 @@
 
 &usbdrd3_0 {
 	status = "okay";
-	extcon = <&fusb0>;
 };
 
 &usbdrd3_1 {
@@ -1037,6 +1036,7 @@
 
 &usbdrd_dwc3_0 {
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usbdrd_dwc3_1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi
index ca07f6032200..c375b9287ff1 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-gru.dtsi
@@ -627,12 +627,12 @@ ap_i2c_audio: &i2c8 {
 
 &usbdrd3_0 {
 	status = "okay";
-	extcon = <&usbc_extcon0>;
 };
 
 &usbdrd_dwc3_0 {
 	status = "okay";
 	dr_mode = "host";
+	extcon = <&usbc_extcon0>;
 };
 
 &vopb {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
index bdeeb9739ba7..497c98f76e3b 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-mid-818-android.dts
@@ -905,12 +905,12 @@
 };
 
 &usbdrd3_0 {
-	extcon = <&fusb0>;
 	status = "okay";
 };
 
 &usbdrd_dwc3_0 {
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &pwm0 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock960-ab.dts b/arch/arm64/boot/dts/rockchip/rk3399-rock960-ab.dts
index a5ea9efd9872..03a364fa4ec5 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rock960-ab.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock960-ab.dts
@@ -710,13 +710,13 @@
 };
 
 &usbdrd3_0 {
-	extcon = <&fusb0>;
 	status = "okay";
 };
 
 &usbdrd_dwc3_0 {
 	dr_mode = "otg";
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usbdrd3_1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rockpro64.dts b/arch/arm64/boot/dts/rockchip/rk3399-rockpro64.dts
index 700dec8e8d19..12ee53859e7d 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rockpro64.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rockpro64.dts
@@ -808,13 +808,13 @@
 };
 
 &usbdrd3_0 {
-	extcon = <&fusb0>;
 	status = "okay";
 };
 
 &usbdrd_dwc3_0 {
 	dr_mode = "otg";
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usbdrd3_1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
index b4b2705f81bf..45a71de1913c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
@@ -539,14 +539,6 @@
 	status = "okay";
 };
 
-&tcphy0 {
-	status = "okay";
-};
-
-&tcphy1 {
-	status = "okay";
-};
-
 &tcphy0 {
 	extcon = <&fusb0>;
 	status = "okay";
@@ -599,7 +591,6 @@
 
 &uart2 {
 	status = "okay";
-	extcon = <&fusb0>;
 };
 
 &usb_host0_ehci {
@@ -625,6 +616,7 @@
 &usbdrd_dwc3_0 {
 	status = "okay";
 	dr_mode = "otg";
+	extcon = <&fusb0>;
 };
 
 &usbdrd3_1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi
index fb7dc0455374..c066349f7b29 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi
@@ -957,12 +957,12 @@
 };
 
 &usbdrd3_0 {
-	extcon = <&fusb0>;
 	status = "okay";
 };
 
 &usbdrd_dwc3_0 {
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &vopb {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts b/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
index 3a4d50da6f8a..c92d8e5f61f3 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tve1205g.dts
@@ -1020,12 +1020,12 @@
 };
 
 &usbdrd3_0 {
-	extcon = <&fusb0>;
 	status = "okay";
 };
 
 &usbdrd_dwc3_0 {
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usb_host1_ehci {
@@ -1037,12 +1037,12 @@
 };
 
 &usbdrd3_1 {
-	extcon = <&fusb1>;
 	status = "okay";
 };
 
 &usbdrd_dwc3_1 {
 	status = "okay";
+	extcon = <&fusb1>;
 };
 
 &rockchip_suspend {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
index 95bda02007f0..fe07e84c72cc 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
@@ -1026,7 +1026,6 @@
 };
 
 &usbdrd3_0 {
-	extcon = <&fusb0>;
 	status = "okay";
 };
 
@@ -1036,6 +1035,7 @@
 
 &usbdrd_dwc3_0 {
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usbdrd_dwc3_1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10.dts
index 49fea10c1cb1..169a64a89124 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10.dts
@@ -918,7 +918,6 @@
 };
 
 &usbdrd3_0 {
-	extcon = <&fusb0>;
 	status = "okay";
 };
 
@@ -928,6 +927,7 @@
 
 &usbdrd_dwc3_0 {
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usbdrd_dwc3_1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
index bec8e80ef6c3..5e200e0f67ba 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
@@ -1032,7 +1032,6 @@
 };
 
 &usbdrd3_0 {
-	extcon = <&fusb0>;
 	status = "okay";
 };
 
@@ -1042,6 +1041,7 @@
 
 &usbdrd_dwc3_0 {
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usbdrd_dwc3_1 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11.dts
index a54ab54c26b5..c90c58d742ab 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11.dts
@@ -926,7 +926,6 @@
 };
 
 &usbdrd3_0 {
-	extcon = <&fusb0>;
 	status = "okay";
 };
 
@@ -936,6 +935,7 @@
 
 &usbdrd_dwc3_0 {
 	status = "okay";
+	extcon = <&fusb0>;
 };
 
 &usbdrd_dwc3_1 {
-- 
2.35.3

